### PR TITLE
Upgrade to ppx_deriving 5.0 (which now uses ppxlib internally)

### PR DESCRIPTION
**Pull request by kit-ty-kate**

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Supersedes https://github.com/ocaml-ppx/ppx_deriving_protobuf/pull/30

File: src/ppx_deriving_protobuf.cppo.ml [hunk 1]
```diff
@@ -509,15 +491,15 @@ let rec derive_reader base_path fields ptype =
...
-    | { pbf_kind = Pbk_repeated } as field :: rest ->
+    | { pbf_kind = Pbk_repeated; pbf_loc = loc } as field :: rest ->
       [%expr let [%p pvar field.pbf_name] = ref [] in [%e mk_cells rest k]]
     | [] -> k
   in
-  let rec mk_reader ({ pbf_name; pbf_path; pbf_enc; pbf_type; } as field) =
+  let rec mk_reader ({ pbf_name; pbf_path; pbf_enc; pbf_type; pbf_loc = loc } as field) =
     let value =
-      let ident = Exp.ident (lid ("Protobuf.Decoder." ^ (string_of_pb_encoding pbf_enc))) in
+      let ident = Exp.ident (Ast_builder.Default.Located.lident ~loc ("Protobuf.Decoder." ^ (string_of_pb_encoding pbf_enc))) in
```
Comment (gasche) [inline src/ppx_deriving_protobuf.cppo.ml:502]: could you make a local definition

```ocaml
let lid ~loc s = Ast_builder.Default.Located.lident ~loc s
```

? Currently several lines are larger-than-my-screen after the change.

File: src/ppx_deriving_protobuf.cppo.ml [hunk 2]
```diff
@@ -94,7 +92,7 @@ let filter_map f lst =
   List.map f lst |> filter []
 
 let string_of_lident lid =
-  String.concat "." (Longident.flatten lid)
+  String.concat "." (Longident.flatten_exn lid)
```
Comment (gasche) [inline src/ppx_deriving_protobuf.cppo.ml:95]: Could you use this helper function in the many other places that (you had to change and) use `String.concat "."` in the same way?

**Review by gasche — APPROVED** [review]
Thanks a lot! I left two small comments for minor changes that would improve the readability of the result, but this is a very nice change.

File: src/ppx_deriving_protobuf.cppo.ml (hunk 1 above)
Comment (kit-ty-kate) [reply src/ppx_deriving_protobuf.cppo.ml]: done

File: src/ppx_deriving_protobuf.cppo.ml (hunk 2 above)
Comment (kit-ty-kate) [reply src/ppx_deriving_protobuf.cppo.ml:98]: done

**Comment by kit-ty-kate** [comment]
Requires https://github.com/ocaml-ppx/ppx_deriving/pull/248

**Comment by gasche** [comment]
Out of curiosity, can you show the place that uses lid-of-string with a dotted path string?

**Comment by kit-ty-kate** [comment]
Sorry for the lack of reply. Here is the test failure I get with plain unpatched `ppx_deriving.5.2`:
```
File "src_test/protoc_inner.ml", lines 1-2, characters 0-32:
1 | type foo = int
2 | [@@deriving protobuf { protoc }]
Error: This variant pattern is expected to have type Protobuf.payload_kind
       The constructor Protobuf.Varint does not belong to type Protobuf.payload_kind
File "src_test/test_syntax.ml", line 15, characters 0-35:
15 | type b = bool [@@deriving protobuf]
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: This variant pattern is expected to have type Protobuf.payload_kind
       The constructor Protobuf.Varint does not belong to type Protobuf.payload_kind
```
The `Protobuf.Varint` string "parsed" as one indent and not `Ldot ("Protobuf", Lident "Varint")` comes from `string_payload_kind_of_pb_encoding` and is passed to `Ppx_deriving.Ast_conveinance.lid_of_string` through `Ppx_deriving.Ast_conveinance.pconstr`.

This function has type `val pconstr : string -> pattern list -> pattern` so one would expect to be able to pass names including its full module path.

**Comment by kit-ty-kate** [comment]
This PR is now ready to be merged (I don't have write access on this one)

**Comment by gasche** [comment]
Thanks! The CI fails on 4.03 and 4.04, and you added a bound (>= 4.05) anyway. Could you change the CI setup to stop testing <4.05?

**Comment by kit-ty-kate** [comment]
Should we simply remove Travis-CI entirely since OCaml-CI is already a superset of Travis?

**Comment by kit-ty-kate** [comment]
CI is green now.

**Comment by gasche** [comment]
The reasoning is that travis-CI was unnecessary because ocaml-ci covers the same checks. Is that correct?

How is the ocaml-ci configured? I couldn't find a configuration file in the repository. (I thought that the tool is still experimental and people need to be whitelisted to use it, does it work because `ocaml-ppx` is whitelisted?)

**Comment by kit-ty-kate** [comment]
> The reasoning is that travis-CI was unnecessary because ocaml-ci covers the same checks. Is that correct?

Yes

> How is the ocaml-ci configured? I couldn't find a configuration file in the repository. (I thought that the tool is still experimental and people need to be whitelisted to use it, does it work because `ocaml-ppx` is whitelisted?)

It's a no-configuration CI. To be eligible for it, the project must be a standard dune project. Internally it calls `dune build @install @runtest` and uses the opam file to install the dependencies.

**Comment by gasche** [comment]
Merged, thanks for the extra explanation. I guess this needs a release now?

**Comment by kit-ty-kate** [comment]
that'd be perfect yes (i can do it for you if you don't have the time, though I'd need write access)

**Comment by gasche** [comment]
Thanks! I just sent you an invitation for maintenance access.

**Pull request merged**